lib/SiftClient.php

```
@@ -2,130 +2,350 @@
+     *     - string 'version': By default, this client's version is used.
+     *     - string 'path': The URL path to use for this call.  By default, the path for the requested
+     *           version of the Events API is used.
+     *
      * @return null|SiftResponse
      */
-    public function track($event, $properties, $timeout = self::DEFAULT_TIMEOUT, $path = null, $returnScore = FALSE, $returnAction = FALSE) {
+    public function track($event, $properties, $opts = array()) {
+        $this->mergeArguments($opts, array(
+            'return_score' => false,
+            'return_action' => false,
+            'return_workflow_status' => false,
+            'abuse_types' => array(),
+            'path' => NULL,
+            'timeout' => $this->timeout,
+            'version' => $this->version
+        ));
         $this->validateArgument($event, 'event', 'string');
         $this->validateArgument($properties, 'properties', 'array');
 
-        if (!$path) $path = self::restApiUrl($returnScore, $returnAction);
+        $path = $opts['path'];
+        if (!$path) {
+            $path = self::restApiUrl($opts['version']);
+        }
+
         $properties['$api_key'] = $this->api_key;
         $properties['$type'] = $event;
+
+        $params = array();
+        if ($opts['return_score']) $params['return_score'] = 'true';
+        if ($opts['return_action']) $params['return_action'] = 'true';
+        if ($opts['return_workflow_status']) $params['return_workflow_status'] = 'true';
+        if ($opts['abuse_types']) $params['abuse_types'] = implode(',', $opts['abuse_types']);
+
         try {
-            $request = new SiftRequest($path, SiftRequest::POST, $properties, $timeout);
+            $request = new SiftRequest(
+                $path, SiftRequest::POST, $opts['timeout'], $opts['version'], array(
+                    'body' => $properties,
+                    'params' => $params
+                ));
```

Longer term, I think we may want to change sift-php so that it throws useful exceptions instead of returning `null` on error.